codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Connection.java

```diff
@@ -277,6 +278,23 @@ public boolean visit(Http2Stream stream) {
         }
     }
 
+    @Override
+    public void headersSent(int streamId, boolean sent) {
+        Http2Stream stream = stream(streamId);
+        assert stream != null : "Invalid stream id";
+        stream.setProperty(headersSentKey, sent);
+    }
+
+    @Override
+    public boolean headersSent(int streamId) {
+        Http2Stream stream = stream(streamId);
+        if (stream == null) {
+            return false;
```

Comment:
Not sure if we want the interface to account for this but just because we don't have a stream object doesn't mean we haven't already sent the headers. For example we could have already sent headers and closed the stream. One approach could be if the stream ID is `< nextStreamIdToCreate` for the particular endpoint we return `true`, and document this behavior in the interface.